mesonbuild/optinterpreter.py

```diff
@@ -169,7 +169,10 @@ def func_option(self, args: T.Tuple[str], kwargs: 'FuncOptionArgs') -> None:
         parser = self.option_types.get(opt_type)
         if not parser:
             raise OptionException(f'Unknown type {opt_type}.')
-        description = kwargs['description'] or opt_name
+        description = kwargs['description']
+
+        if not description and not isinstance(kwargs['deprecated'], str):
+            description = opt_name
```

Comment:
I'm not sure it is ever useful to repeat the option name as description. Maybe we should always leave it empty?

Reply:
I agree, but that was a documented behavior, so I was conservative here. I've changed this in a separate commit now and updated the docs.